data/block/block.go

```diff
@@ -581,3 +582,39 @@ func (h *Header) GetAdditionalData() headerVersionData.HeaderAdditionalData {
 	// no extra data for the initial version of shard block header
 	return nil
 }
+
+// CheckFieldsForNil checks a predefined set of fields for nil values
+func (h *Header) CheckFieldsForNil() error {
```

Comment:
are all these checks backwards compatible? did we have some of them nil sometimes and now they cannot be nil?

Reply:
will definitely check with an import-db. At the first sight, they should be backwards compatible

Reply:
ok